### PR TITLE
Remove agent relate stackdriver logging and taskattempt code from app_dart

### PR DESCRIPTION
**Pull request by yusufm**
Remove stackdriver logging and task attempt summary code which was used by now defunct agent code.

TEST=app_flutter/flutter test
TEST=app_dart/dart test

### CODE REVIEW

File: app_flutter/lib/widgets/task_overlay.dart [hunk 1]
```diff
@@ -291,7 +290,6 @@ class TaskOverlayContents extends StatelessWidget {
                               'Task was run outside of devicelab',
                               style: Theme.of(context).textTheme.bodyText2,
                             ),
-                          if (QualifiedTask.fromTask(task).isDevicelab) TaskAttemptSummary(task: task),
                           if (QualifiedTask.fromTask(task).isLuci) LuciTaskAttemptSummary(task: task),
```
Comment (CaseyHillers) [inline app_flutter/lib/widgets/task_overlay.dart:293]: Can we rename `LuciTaskAttemptSummary` to `TaskAttemptSummary` now?

Comment (yusufm) [reply app_flutter/lib/widgets/task_overlay.dart:293]: I'm leaning to leaving it as is for simplicities sake, unless there is a compelling reason to rename.

Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_overlay.dart:293]: With the code cleanup from last week all tasks are `LuciTask`, so I think it adds TD to leave this naming here (how does `LuciTask` differ from `Task`). I'll defer to @godofredoc for what we should do.

Comment (yusufm) [reply app_flutter/lib/widgets/task_overlay.dart:293]: With your changes to remove the stage concept, is QualifiedTask going away as well?

Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_overlay.dart:293]: I removed `Stage` from the backend (+ whatever was needed in the frontend). Feel free to remove it if it no longer makes sense.

Comment (yusufm) [reply app_flutter/lib/widgets/task_overlay.dart:293]: .isLuci is based on the stage thats set when creating QualifiedTask. That still seems to be around after your change?

Comment (CaseyHillers) [reply app_flutter/lib/widgets/task_overlay.dart:293]: Yes, that's a frontend relic. My PR should have been labeled "remove stage from backend"

**Review by CaseyHillers — APPROVED** [review]
LGTM - Talked offline, Yusuf to follow up with separate PR with remaining cleanup

**Comment by yusufm** [comment]
Updated goldens for flutter test.

**Pull request merged**